### DISCUSSION TITLE
How recognize PyInstaller was uesd

### DISCUSSION THREAD

**Discussion by emcek** [pyinstaller/pyinstaller#7854] [Help] [answered]
I develop simple application (which can be run both like regular Python app and as PyInstaller executable).

I would like to detect which way user use it. My first guess was: `sys.executable` but it does not give me clear difference.

Any other way?

**Comment by rokm** [comment, marked as answer]
You can check for `sys.frozen`, which is set by PyInstaller: `is_pyinstaller = getattr(sys, 'frozen', False)`. Similarly, you could also check if `sys._MEIPASS` exists using `hasattr(sys, '_MEIPASS')`.
